src/Symfony/Component/Cache/Adapter/OpCacheAdapter.php

```diff
@@ -0,0 +1,287 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Adapter;
+
+use Psr\Cache\CacheItemInterface;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerAwareTrait;
+use Symfony\Component\Cache\CacheItem;
+use Symfony\Component\Cache\Exception\InvalidArgumentException;
+
+/**
+ * Adapter building static PHP files that will be cached by OPCache.
+ * This Adapter is in read-only if you use AdapterInterface methods.
+ * You can use the method "store" to build the cache file.
+ *
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ */
+class OpCacheAdapter implements AdapterInterface, LoggerAwareInterface
+{
+    use LoggerAwareTrait;
+
+    private $file;
+    private $storeSerialized;
+    private $values;
+    private $createCacheItem;
+
+    public function __construct($file, $storeSerialized = true)
+    {
+        if (file_exists($file)) {
+            if (!is_file($file)) {
+                throw new InvalidArgumentException(sprintf('Cache path exists and is not a file: %s', $file));
+            }
+
+            if (!is_writable($file)) {
+                throw new InvalidArgumentException(sprintf('Cache file is not writable: %s', $file));
+            }
+        }
+
+        $directory = dirname($file);
+
+        if (!file_exists($directory)) {
+            @mkdir($directory, 0777, true);
+        }
+
+        if (!file_exists($directory)) {
+            throw new InvalidArgumentException(sprintf('Cache directory does not exist and cannot be created: %s', $directory));
+        }
+
+        if (!is_writable($directory)) {
+            throw new InvalidArgumentException(sprintf('Cache directory is not writable: %s', $directory));
+        }
+
+        $this->file = $file;
+        $this->storeSerialized = $storeSerialized;
+
+        $this->createCacheItem = \Closure::bind(
+            function ($key, $value, $isHit) {
+                $item = new CacheItem();
+                $item->key = $key;
+                $item->value = $value;
+                $item->isHit = $isHit;
+
+                return $item;
+            },
+            $this,
+            CacheItem::class
+        );
+    }
+
+    public function store(array $values = array())
```

Comment:
I think we should add an authoritative mode by adding a `$authoritativeDefaultValue` as 2nd arg, that this adapter returns instead of cache misses.